src/Net.jl

```diff
@@ -112,12 +113,51 @@ function chread(
     data
 end
 
+Base.collect(t::Union{Type, DataType, Union{}}) = _collect(t, [])
+_collect(t::Type, list) = t<:Union{} ? push!(list, t) : _collect(t.b, push!(list, t.a))
+_collect(t::Union{DataType,Core.TypeofBottom}, list) = push!(list, t)
+
+function chread(
+    sock::ClickHouseSock,
+    ::Type{Vector{T}},
+    count::VarUInt,
+)::Vector{T} where T <: Union{Missing, Number}
+    type = (T -> for typeName in collect(T)
+                    if typeName !== Missing
+                        return typeName
+                    end
+                end
+    )(T)
+
+    isNull = Vector{Int8}(undef, UInt64(count))
+    data = Vector{type}(undef, UInt64(count))
+    read!(sock.io, isNull)
+    read!(sock.io, data)
+    data = convert(Array{T}, data)
+    map!(data, data, isNull) do x, flag
+        if flag == 1
+            return missing
+        else
+            return x
+        end
+    end
+    return data
+end
+
 chread(
     sock::ClickHouseSock,
     ::Type{Vector{String}},
     count::VarUInt,
 )::Vector{String} = [chread(sock, String) for _ ∈ 1:UInt64(count)]
 
+struct FixedString
+    length::Int
+end
+
+function chread(sock, ty::FixedString, count)::Vector{String}
+    [chread(sock, UInt64(ty.length)) |> String for _ ∈ 1:UInt64(count)]
+end
+
```

Comment:
This has a `chread` implementation but lacks a `chwrite` one, breaking select/insertion symmetry. Please also implement `chwrite` for this.